python/langsmith/client.py

```diff
@@ -1212,18 +1215,34 @@ def create_run(
         }
         if not self._filter_for_sampling([run_create]):
             return
-        run_create = self._run_transform(run_create, copy=True)
-        if revision_id is not None:
-            run_create["extra"]["metadata"]["revision_id"] = revision_id
+
         if (
             self.tracing_queue is not None
             # batch ingest requires trace_id and dotted_order to be set
             and run_create.get("trace_id") is not None
             and run_create.get("dotted_order") is not None
         ):
+            attachments_collector: Dict[str, ls_schemas.Attachments] = {}
+            run_create = self._run_transform(
+                run_create,
+                copy=False,
+                attachments_collector=attachments_collector,
+            )
+            if revision_id is not None:
+                run_create["extra"]["metadata"]["revision_id"] = revision_id
+            acc = convert_to_multipart_parts_and_context(
+                [run_create], [], all_attachments=attachments_collector
+            )
```

Comment:
this is the "new stuff" in the hot path - except we have a massive removal by doing `_run_transform(copy=False)` instead of `copy=True`